android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt

```diff
@@ -105,19 +109,28 @@ class CreateListingPageViewModel @Inject constructor(
 						}
 					}.awaitAll()
 
+					val location = locationService.getLocation()
 					listingsApi.listingsCreate(
 						CreateListingRequest(
 							addressLine = it.address.addressLine,
 							addressCity = it.address.addressCity,
 							addressPostalcode = it.address.addressPostalCode,
 							addressCountry = it.address.addressCountry,
 							price = it.price,
-							rooms = it.numBedrooms,
 							leaseStart = it.startDate,
 							leaseEnd = it.endDate,
 							description = it.description,
 							residenceType = ResidenceType.house,
 							imgIds = imgIds,
+							latitude = location!!.latitude.toFloat(),
+							longitude = location.longitude.toFloat(),
```

Comment:
why does this line not have `!!`?

Reply:
I think if the previously executed location!! succeeds, then kotlin can tell that location is non-null and will compile.

Reply:
Yeah, Kotlin has "smart-casting" which allows Kotlin to deduce that if line 125 doesn't destroy the app, then location must not be null

Reply:
I see 
if it runs then no issue